### PR TITLE
Only sign in the official build pipeline, not in any other pipeline

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Use the unused `signBinaries` parameter to control when we sign so we only sign during the official build.

Official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=886183&view=results

Perf build: https://dev.azure.com/dnceng/internal/_build/results?buildId=886184&view=results

### CODE REVIEW

File: eng/pipelines/coreclr/templates/build-job.yml [hunk 1]
```diff
@@ -231,7 +231,7 @@ jobs:
             overWrite: true
 
     # Sign diagnostic files on Windows
-    - ${{ if and(eq(parameters.osGroup, 'windows'), ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), ne(parameters.testGroup, 'clrTools')) }}:
```
Comment (hoyosjs) [inline eng/pipelines/coreclr/templates/build-job.yml:234]: I believe `clrTools` was needed here for something that David added. Can't recall if it was managed PGO or CrossGen2

Comment (hoyosjs) [reply eng/pipelines/coreclr/templates/build-job.yml:234]: Also, if this is to fix the build perf pipeline (sorry, I was out) the same check should be done on line 135

**Review by hoyosjs — APPROVED** [review]

File: eng/pipelines/coreclr/templates/build-job.yml (hunk 1 above)
Comment (hoyosjs) [reply eng/pipelines/coreclr/templates/build-job.yml:234]: Last point was solved with your last commit.

Comment (jkoritzinsky) [reply eng/pipelines/coreclr/templates/build-job.yml:234]: Clrtools is only used in the public build. We don't pass down clrtools as a test group in the official build

Comment (jkoritzinsky) [reply eng/pipelines/coreclr/templates/build-job.yml:234]: I've made the equivalent fix on 135 as well.

**Comment by jkoritzinsky** [comment]
The perf build and the official build have gotten past the CoreCLR windows x64 builds, so this looks good to me. I'll mark this with auto-merge unless we want to merge this before CI finishes.

**Review by safern — APPROVED** [review]

**Comment by DrewScoggins** [comment]
LGTM

**Review by DrewScoggins — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
This PR only has changes on Windows, so the failure is unrelated.

**Pull request merged**